### PR TITLE
{bio}[foss/2023a] ColabFold v1.5.5 w/ CUDA 12.1.1

### PR DESCRIPTION
**Pull request by ThomasHoffmann77**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
@ThomasHoffmann77 Can you clarify why this is marked as draft/WIP?

**Comment by ThomasHoffmann77** [comment]
> @ThomasHoffmann77 Can you clarify why this is marked as draft/WIP?

I was waiting for feedback from our users and for  a new version tag 1.5.6

File: easybuild/easyconfigs/c/ColabFold/ColabFold-1.5.5-foss-2023a-CUDA-12.1.1.eb [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+            'AlphaFold-2.3.2_data-dep-paths-shebang-UniRef30.patch',
+            'AlphaFold-2.0.0_n-cpu.patch',
+            'AlphaFold-2.0.1_setup_rm_tfcpu.patch',
+            'AlphaFold-2.3.2_BioPythonPDBData.patch',
+        ],
+        'preinstallopts': " cp %(builddir)s/stereo_chemical_props-7102c6.txt " +
+                          "alphafold/common/stereo_chemical_props.txt &&",
+        'runtest': "python ./run_alphafold_test.py",
+        'source_urls': [
+            'https://github.com/sokrypton/alphafold/archive/',
```
Comment (boegel) [inline easybuild/easyconfigs/c/ColabFold/ColabFold-1.5.5-foss-2023a-CUDA-12.1.1.eb:104]: @ThomasHoffmann77 I think it makes sense to add a comment here to explain why we're using a fork of `AlphaFold` here and bake it into the `ColabFold` installation, rather than just including `AlphaFold` as a dependency..

File: easybuild/easyconfigs/c/ColabFold/ColabFold-1.5.5-foss-2023a-CUDA-12.1.1.eb [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+            'https://git.scicore.unibas.ch/schwede/openstructure/-/raw/7102c6/modules/mol/alg/src/',
```
Comment (boegel) [inline easybuild/easyconfigs/c/ColabFold/ColabFold-1.5.5-foss-2023a-CUDA-12.1.1.eb:105]: Why is this required?

I think this is only necessary for `stereo_chemical_props.txt`, which is handled via `components`, so this line can be removed?

```suggestion
```

File: easybuild/easyconfigs/c/ColabFold/ColabFold-1.5.5-foss-2023a-CUDA-12.1.1.eb [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+    }),
+    ('contextlib2', '21.6.0', {
+        'checksums': ['ab1e2bfe1d01d968e1b7e8d9023bc51ef3509bba217bb730cee3827e1ee82869'],
+    }),
+    ('ml_collections', '0.1.1', {
+        'preinstallopts': "touch requirements.txt && touch requirements-test.txt && ",
+        'checksums': ['3fefcc72ec433aa1e5d32307a3e474bbb67f405be814ea52a2166bfc9dbe68cc'],
+    }),
+    ('AlphaFold', '2.3.6', {
+        'easyblock': 'PythonPackage',
```
Comment (boegel) [inline easybuild/easyconfigs/c/ColabFold/ColabFold-1.5.5-foss-2023a-CUDA-12.1.1.eb:93]: There's no custom easyblock for `AlphaFold`, so no need for this line?
```suggestion
```

File: easybuild/easyconfigs/c/ColabFold/ColabFold-1.5.5-foss-2023a-CUDA-12.1.1.eb [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        'checksums': ['93d100ccd2cd09a1fd3f136b9328c6e59529ba341de8bb499437f6819159fe8a'],
+    }),
+    ('contextlib2', '21.6.0', {
+        'checksums': ['ab1e2bfe1d01d968e1b7e8d9023bc51ef3509bba217bb730cee3827e1ee82869'],
+    }),
+    ('ml_collections', '0.1.1', {
+        'preinstallopts': "touch requirements.txt && touch requirements-test.txt && ",
+        'checksums': ['3fefcc72ec433aa1e5d32307a3e474bbb67f405be814ea52a2166bfc9dbe68cc'],
+    }),
+    ('AlphaFold', '2.3.6', {
```
Comment (boegel) [inline easybuild/easyconfigs/c/ColabFold/ColabFold-1.5.5-foss-2023a-CUDA-12.1.1.eb:92]: There are no tags in https://github.com/sokrypton/alphafold/, so I don't think we should invent a version here.

Just use datestamp of the commit, maybe even with commit as extra label?

```suggestion
    ('AlphaFold', '20230725-%s' % _sokrypton_commit, {

File: easybuild/easyconfigs/c/ColabFold/ColabFold-1.5.5-foss-2023a-CUDA-12.1.1.eb [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    "colabfold_batch --help",
+]
+
+sanity_pip_check = True
+
+modextravars = {
+    'TF_FORCE_UNIFIED_MEMORY': '1',
+    'XLA_PYTHON_CLIENT_MEM_FRACTION': '3',
+    # 'ALPHAFOLD_DATA_DIR': '/path/to/AlphaFold_DBs/2.3.2',  # please adapt
+    'ALPHAFOLD_DATA_DIR': '/scratch/AlphaFold_DBs/2.3.2',  # please adapt,
```
Comment (boegel) [inline easybuild/easyconfigs/c/ColabFold/ColabFold-1.5.5-foss-2023a-CUDA-12.1.1.eb:169]: I would remove this line, and just not set `$ALPHAFOLD_DATA_DIR` by default, we can't use any sensible default value here anyway...

```suggestion
```

File: easybuild/easyconfigs/c/ColabFold/ColabFold-1.5.5-foss-2023a-CUDA-12.1.1.eb [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+
+modloadmsg = """
+%(name)s notebooks are located in $EBROOTCOLABFOLD/notebooks
+"""
+
+postinstallmsgs = [
+    "A newer version of UniRef30 (2023_02) is available at: "
+    "https://wwwuser.gwdg.de/~compbiol/uniclust/2023_02/UniRef30_2023_02_hhsuite.tar.gz. "
+    "Untar to $ALPHAFOLD_DATA_DIR/uniref30/ and set the default version accordingly by changing "
+    "modextravars:ALPHAFOLD_UNIREF30_VER."
```
Comment (boegel) [inline easybuild/easyconfigs/c/ColabFold/ColabFold-1.5.5-foss-2023a-CUDA-12.1.1.eb:184]: `modextravars` is meaningless for end users, I would rephrase this to talk about changing the "`$ALPHAFOLD_UNIREF30_VER` environment variable" ?

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/c/ColabFold/ColabFold-1.5.5-foss-2023a-CUDA-12.1.1.eb (hunk 6 above)
Comment (ThomasHoffmann77) [reply easybuild/easyconfigs/c/ColabFold/ColabFold-1.5.5-foss-2023a-CUDA-12.1.1.eb:181]: @boegel Do "end users" get this message? I assumed `postinstallmsgs` is intended for admins.

**Comment by verdurin** [comment]
@ThomasHoffmann77 do you consider this ready for review now?

**Comment by ThomasHoffmann77** [comment]
> @ThomasHoffmann77 do you consider this ready for review now?

I'd say yes.
Not sure, whether the modextravars part requires some more attention. I'd keep it as it is.

**Comment by verdurin** [comment]
I think the point is just to mention the environment variable itself - `modextravars` is not a word that will be meaningful to users, given it's very unlikely they will look inside an easyconfig. I'll add a comment in the review.

File: easybuild/easyconfigs/c/ColabFold/ColabFold-1.5.5-foss-2023a-CUDA-12.1.1.eb (hunk 6 above)
Comment (ThomasHoffmann77) [reply easybuild/easyconfigs/c/ColabFold/ColabFold-1.5.5-foss-2023a-CUDA-12.1.1.eb:182]: @boegel @verdurin Should I just remove the postinstallmsg? There is still the comment explaining how to use ALPHAFOLD_UNIREF30_VER if required.

Comment (verdurin) [reply easybuild/easyconfigs/c/ColabFold/ColabFold-1.5.5-foss-2023a-CUDA-12.1.1.eb:182]: @verdurin that might be simpler - retain the information as a comment.

Comment (ThomasHoffmann77) [reply easybuild/easyconfigs/c/ColabFold/ColabFold-1.5.5-foss-2023a-CUDA-12.1.1.eb]: Postinstallmsg deleted 
@verdurin @boegel

File: easybuild/easyconfigs/c/ColabFold/ColabFold-1.5.5-foss-2023a-CUDA-12.1.1.eb [hunk 7]
```diff
@@ -175,11 +175,4 @@ modloadmsg = """
 %(name)s notebooks are located in $EBROOTCOLABFOLD/notebooks
 """
 
-postinstallmsgs = [
```
Comment (verdurin) [inline easybuild/easyconfigs/c/ColabFold/ColabFold-1.5.5-foss-2023a-CUDA-12.1.1.eb]: @ThomasHoffmann77 apologies for this extra hassle, but I thought you were going to keep this text and put it as a comment? Seems like valuable information.

Comment (ThomasHoffmann77) [reply easybuild/easyconfigs/c/ColabFold/ColabFold-1.5.5-foss-2023a-CUDA-12.1.1.eb]: Thought the comments in line 169 were enough. However, I'll add some more infos tomorrow
